File: Data Structures & Algorithms/is-anagram/submission-1.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        a = {}
        for cs in s:
            if cs in a:
                a[cs] += 1
            else:
                a[cs] = 1
        for ct in t:
            if ct not in a or a[ct] == 0:
                return False
            else:
                a[ct] -= 1
        return True

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        // Optimally initializes the set in one line, skipping the manual loop
        unordered_set<int> numset(nums.begin(), nums.end());
        int res = 0;
        
        for (int num : numset) {
            // If num-1 doesn't exist, 'num' is the start of a sequence
            if (!numset.count(num - 1)) {
                int count = 1;
                
                // Count upwards directly without a separate 'val' variable
                while (numset.count(num + count)) {
                    count++;
                }
                
                res = max(res, count);
            }
        }
        
        return res;
    }
};